tests/src/test/java/com/tradeshift/scalajapi/akka/route/RouteTest.java

```diff
@@ -0,0 +1,288 @@
+                    .map(s -> new BigDecimal(s));
+    
+    private final Unmarshaller<HttpRequest,UUID> UUID_FROM_JSON_BODY =
+        Unmarshaller.requestToEntity()
+                    .flatMap(
+                        Unmarshaller.forMediaType(MediaTypes.APPLICATION_JSON,
+                        Unmarshaller.entityToString(materializer())))
+                    .map(s -> {
+                        // just a fake JSON parser, assuming it's {"id":"..."}
+                        Pattern regex = Pattern.compile("\"id\":\"(.+)\"");
+                        Matcher matcher = regex.matcher(s);
+                        matcher.find();
+                        return UUID.fromString(matcher.group(1));
+                    });
+    
+    private final Unmarshaller<HttpRequest,UUID> UUID_FROM_XML_BODY =
+        Unmarshaller.requestToEntity()
+                    .flatMap(
+                        Unmarshaller.forMediaTypes(Seq.of(MediaTypes.TEXT_XML, MediaTypes.APPLICATION_XML),
+                        Unmarshaller.entityToString(materializer())))
+                    .map(s -> {
+                        // just a fake XML parser, assuming it's <id>...</id>
+                        Pattern regex = Pattern.compile("<id>(.+)</id>");
+                        Matcher matcher = regex.matcher(s);
+                        matcher.find();
+                        return UUID.fromString(matcher.group(1));
+                    });
+    
+    private final Unmarshaller<HttpRequest,UUID> UUID_FROM_BODY = 
+        Unmarshaller.firstOf(UUID_FROM_JSON_BODY, UUID_FROM_XML_BODY);
+    
+    private final Marshaller<UUID, RequestEntity> UUID_TO_RQ = Marshaller.wrapEntity(
+        Marshaller.stringToEntity(), 
+        ContentType.create(MediaTypes.APPLICATION_JSON), 
+        (UUID u) -> "{\"id\":\"" + u + "\"}");
+    
+    private final Marshaller<UUID, HttpResponse> UUID_TO_JSON_BODY =
+        Marshaller.entityToResponse(
+        Marshaller.wrapEntity(Marshaller.stringToEntity(), ContentType.create(MediaTypes.APPLICATION_JSON), 
+            (UUID u) -> "{\"id\":\"" + u + "\"}"));
+    
+    private final Marshaller<UUID, HttpResponse> UUID_TO_XML_BODY(MediaType xmlType) { return
+        Marshaller.entityToResponse(
+        Marshaller.wrapEntity(Marshaller.byteStringToEntity(), ContentType.create(xmlType), 
+            (UUID u) -> ByteString.fromString("<id>" + u + "</id>")));
+    }
+    
+    private final Marshaller<UUID, HttpResponse> UUID_TO_BODY = Marshaller.oneOf(
+        UUID_TO_JSON_BODY,
+        UUID_TO_XML_BODY(MediaTypes.APPLICATION_XML),
+        UUID_TO_XML_BODY(MediaTypes.TEXT_XML));
+    
+    @Test
+    public void path_can_match_uuid() {
+        on(HttpRequest.GET("/documents/359e4920-a6a2-4614-9355-113165d600fb"), route, () -> {
+            assertThat(responseEntityStrict().data().utf8String()).isEqualTo("document 359e4920-a6a2-4614-9355-113165d600fb");            
```

Comment:
That's not a typo. The test runs green. The response is expected to indicate _document number [uuid]_. Hence, single instead of plural.